libs/payments/stripe/src/lib/stripe.manager.ts

```diff
@@ -74,4 +81,49 @@ export class StripeManager {
       customer.tax?.automatic_tax === 'not_collecting'
     );
   }
+
+  /**
+   * Updates customer object with incoming tax ID if existing tax ID does not match
+   *
+   * @param customerId Customer ID of customer to be updated
+   * @param taxId Customer tax ID to be updated if different from existing tax ID
+   * @returns Updated customer, or undefined if existing customer tax ID matches
+   */
+  async setCustomerTaxId(customerId: string, taxId: string) {
+    const customerTaxId = await this.getCustomerTaxId(
+      customerId,
+      MOZILLA_TAX_ID
+    );
+
+    if (customerTaxId !== taxId || customerTaxId === '') {
+      return await this.client.updateCustomer(customerId, {
+        invoice_settings: {
+          custom_fields: [{ name: MOZILLA_TAX_ID, value: taxId }],
+        },
+      });
+    }
+
+    return;
+  }
+
+  /**
+   * Returns tax ID of customer
+   *
+   * @param customerId ID of customer to fetch and check existing tax ID
+   * @param taxName Custom field name for given tax ID
+   * @returns The tax ID of customer or empty string if not found
+   */
+  async getCustomerTaxId(customerId: string, taxName: string) {
+    const customer = await this.client.fetchCustomer(customerId);
+    let taxValue = '';
+
+    if (!customer) throw new CustomerNotFoundError();
+    if (customer.deleted) throw new CustomerDeletedError();
+
+    customer.invoice_settings.custom_fields?.map((custom_field: any) => {
+      if (custom_field.name === taxName) taxValue = custom_field.value;
+    });
```

Comment:
Typically `.map` is used for mutating an array, and should not reference values outside of the context of `.map`. Instead, for this purpose either `.forEach` is used (since your .map currently just returns an array of undefined), or a simple for loop is used.

I'd propose this implementation instead:

```
    const customer = await this.client.fetchCustomer(customerId);

    if (!customer) throw new CustomerNotFoundError();
    if (customer.deleted) throw new CustomerDeletedError();

    const customFields = customer.invoice_settings.custom_fields || [];
    const taxIdFields = customFields.filter((custom_field: any) => {
      return custom_field.name === taxName;
    });

    return taxIdFields.at(0)?.value;
```

Note that the above now returns either the user's tax ID or undefined if not present.

Reply:
Nice! Excellent suggestion! I like it. :) Updated!